### PR TITLE
Shim Interop.libc.socket().

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@stephentoub @nguerrera @sokket PTAL

File: src/Native/System.Native/pal_networking.cpp [hunk 1]
```diff
@@ -403,72 +403,102 @@ extern "C" Error GetIPSocketAddressSizes(int32_t* ipv4SocketAddressSize, int32_t
...
+
+    auto* sockAddr = reinterpret_cast<const sockaddr*>(socketAddress);
+    if (!IsInBounds(sockAddr, static_cast<size_t>(socketAddressLen), &sockAddr->sa_family))
+    {
+        return PAL_EFAULT;
+    }
+
+    if (!TryConvertAddressFamilyPlatformToPal(sockAddr->sa_family, addressFamily))
+    {
+        // TODO: we may want to let the address family through transparently if it's
```
Comment (sokket) [inline src/Native/System.Native/pal_networking.cpp]: Super NIT: Is there a bug for this?

**Comment by sokket** [comment]
LGTM, just one question about a TODO

File: src/Native/System.Native/pal_networking.cpp (hunk 1 above)
Comment (pgavlin) [inline src/Native/System.Native/pal_networking.cpp]: There is not--I'll file one.

**Pull request merged**